### PR TITLE
limit interfaces and user privileges [full ci]

### PR DESCRIPTION
**Pull request by hickeng**
~~Run docker as "nobody"~~ the personality is doing port forwarding and via libnetwork/iptables package so no mechanism to sudo or similar just for that operation.

Limit vicadmin and docker to listening on "client" interface
Move sensitive data to env vars instead of args (partial migration to extraconfig)

Fixes: #2632 bug1727644
Towards: #595 #596 #599

### CODE REVIEW

**Comment by hickeng** [comment]
```
vagrant@devbox:~/vic$ vic-ssh
SSH to snip.eng.vmware.com
Warning: Permanently added 'snip.eng.vmware.com,10.136.31.238' (ECDSA) to the list of known hosts.
root@ghicken-test [ ~ ]# ps -ef
UID        PID  PPID  C STIME TTY          TIME CMD
root         1     0  0 01:29 ?        00:00:02 /lib/systemd/systemd
root         2     0  0 01:29 ?        00:00:00 [kthreadd]
root         3     2  0 01:29 ?        00:00:00 [ksoftirqd/0]
root         5     2  0 01:29 ?        00:00:00 [kworker/0:0H]
root         7     2  0 01:29 ?        00:00:00 [rcu_sched]
root         8     2  0 01:29 ?        00:00:00 [rcu_bh]
root         9     2  0 01:29 ?        00:00:00 [migration/0]
root        10     2  0 01:29 ?        00:00:00 [kdevtmpfs]
root        11     2  0 01:29 ?        00:00:00 [netns]
root        12     2  0 01:29 ?        00:00:00 [perf]
root        13     2  0 01:29 ?        00:00:00 [writeback]
root        14     2  0 01:29 ?        00:00:00 [ksmd]
root        15     2  0 01:29 ?        00:00:00 [khugepaged]
root        16     2  0 01:29 ?        00:00:00 [crypto]
root        18     2  0 01:29 ?        00:00:00 [bioset]
root        19     2  0 01:29 ?        00:00:00 [kblockd]
root        20     2  0 01:29 ?        00:00:00 [kswapd0]
root        21     2  0 01:29 ?        00:00:00 [vmstat]
root        22     2  0 01:29 ?        00:00:00 [fsnotify_mark]
root        33     2  0 01:29 ?        00:00:00 [scsi_eh_0]
root        34     2  0 01:29 ?        00:00:00 [scsi_tmf_0]
root        35     2  0 01:29 ?        00:00:00 [vmw_pvscsi_wq_0]
root        36     2  0 01:29 ?        00:00:00 [kworker/u2:1]
root        47     2  0 01:29 ?        00:00:00 [deferwq]
root        55     2  0 01:29 ?        00:00:00 [ipv6_addrconf]
root        77     1  0 01:29 ?        00:00:00 /lib/systemd/systemd-journald
root        84     2  0 01:29 ?        00:00:00 [kworker/0:4]
root       103     1  0 01:29 ?        00:00:00 /lib/systemd/systemd-udevd
systemd+   113     1  0 01:29 ?        00:00:00 /lib/systemd/systemd-timesyncd
systemd+   140     1  0 01:29 ?        00:00:00 /lib/systemd/systemd-resolved
root       145     1  0 01:29 ?        00:00:00 /usr/sbin/haveged -w 1024 -v 1 --Foreground
root       152     2  0 01:29 ?        00:00:00 [kworker/0:6]
root       180     2  0 01:29 ?        00:00:00 [kworker/u2:3]
root       182     1  0 01:29 ?        00:00:02 /sbin/vic-init
root       183     2  0 01:29 ?        00:00:00 [ata_sff]
root       185     1  0 01:29 tty1     00:00:00 /sbin/agetty --noclear tty1 linux
root       187     2  0 01:29 ?        00:00:00 [scsi_eh_1]
root       189     2  0 01:29 ?        00:00:00 [scsi_tmf_1]
root       190     2  0 01:29 ?        00:00:00 [scsi_eh_2]
root       192     2  0 01:29 ?        00:00:00 [scsi_tmf_2]
root       200     2  0 01:29 ?        00:00:00 [kworker/u2:4]
root       204     2  0 01:29 ?        00:00:00 [ttm_swap]
root       205     2  0 01:29 ?        00:00:00 [kpsmoused]
root       207     2  0 01:29 ?        00:00:00 [bioset]
nobody     230   182  0 01:29 ?        00:00:00 /sbin/docker-engine-server -port=2376 -port-layer-port=2377
vicadmin   242   182  0 01:29 ?        00:00:01 /sbin/vicadmin --dc=clovervm --pool=/clovervm/host/bonneville/Resources --cluster=/clovervm/host/bonneville
root       249   182  0 01:30 ?        00:00:01 /sbin/port-layer-server --host=localhost --port=2377
root       265     2  0 01:30 ?        00:00:00 [kworker/0:1H]
root       336     1  0 01:34 ?        00:00:00 /usr/sbin/sshd -D
root       344   336  0 01:34 ?        00:00:00 sshd: root@pts/0
root       347   344  0 01:34 pts/0    00:00:00 -bash
root       354     2  0 01:34 ?        00:00:00 [kworker/0:0]
root       355   347  0 01:34 pts/0    00:00:00 ps -ef
root@ghicken-test [ ~ ]#
```

```
root@ghicken-test [ ~ ]# netstat -an | grep LISTEN
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
tcp        0      0 10.136.31.238:2376      0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:2377          0.0.0.0:*               LISTEN
tcp        0      0 10.136.31.238:2377      0.0.0.0:*               LISTEN
tcp        0      0 10.136.31.238:2378      0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:6060          0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:6061          0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:6062          0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:6063          0.0.0.0:*               LISTEN
tcp6       0      0 :::53                   :::*                    LISTEN
tcp6       0      0 :::22                   :::*                    LISTEN
unix  2      [ ACC ]     STREAM     LISTENING     12646    /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     12660    /run/systemd/journal/stdout
unix  2      [ ACC ]     SEQPACKET  LISTENING     12680    /run/udev/control
root@ghicken-test [ ~ ]#
```

with `--debug=3`
```
root@ghicken-test [ ~ ]# netstat -an | grep LISTEN
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:2377          0.0.0.0:*               LISTEN
tcp        0      0 192.168.78.146:2377     0.0.0.0:*               LISTEN
tcp        0      0 192.168.78.146:2378     0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:6060          0.0.0.0:*               LISTEN
tcp        0      0 192.168.78.146:6061     0.0.0.0:*               LISTEN
tcp        0      0 192.168.78.146:6062     0.0.0.0:*               LISTEN
tcp        0      0 192.168.78.146:6063     0.0.0.0:*               LISTEN
tcp6       0      0 :::53                   :::*                    LISTEN
tcp6       0      0 :::22                   :::*                    LISTEN
tcp6       0      0 :::2376                 :::*                    LISTEN
unix  2      [ ACC ]     STREAM     LISTENING     12646    /run/systemd/private
unix  2      [ ACC ]     SEQPACKET  LISTENING     12661    /run/udev/control
unix  2      [ ACC ]     STREAM     LISTENING     12671    /run/systemd/journal/stdout
```

**Comment by hickeng** [comment]
@gigawhitlocks @sgairo @vburenin please take a look - it's ready to go.

**Comment by hickeng** [comment]
and that last build is what happens when you're local env thinks a file is executable (and builds it as such), but the build system disagrees...

File: cmd/vic-init/ops.go [hunk 1]
```diff
@@ -69,6 +70,9 @@ func (t *operations) HandleSessionExit(config *tether.ExecutorConfig, session *t
 			return
 		}
 
+		// incredibly basic throttle
+		time.Sleep(3 * time.Second)
```
Comment (hmahmood) [inline cmd/vic-init/ops.go]: 3 seconds seems pretty long for throttle. Something sub-second will no do the job here?

File: isos/appliance.sh [hunk 2]
```diff
@@ -103,14 +101,17 @@ chown -R 1000:1000 $(rootfs_dir $PKGDIR)/home/vicadmin
 install -m 755 -d $(rootfs_dir $PKGDIR)/etc/tmpfiles.d
 echo "m  /var/log/journal/%m/system.journal 2755 root systemd-journal - -" > $(rootfs_dir $PKGDIR)/etc/tmpfiles.d/systemd.conf
 
+chroot $(rootfs_dir $PKGDIR) groupadd -g 1001 docker
```
Comment (hmahmood) [inline isos/appliance.sh]: Are we using the docker user anywhere?

File: cmd/vic-init/ops.go (hunk 1 above)
Comment (hickeng) [inline cmd/vic-init/ops.go]: could do with testing in a situation that causes the tight restart loop.
Happy to start this much short if we have some sort of back-off - worst case here we're DoS'ing the infrastructure.

File: isos/appliance.sh (hunk 2 above)
Comment (hickeng) [inline isos/appliance.sh]: nope - will remove it

File: cmd/vicadmin/logs.go [hunk 3]
```diff
@@ -154,15 +154,12 @@ func configureReaders() map[string]entryReader {
 		readers[path[1:]] = fileReader(path)
 	}
 
-	if config.vmPath == "" {
-		log.Info("vm-path not set, skipping datastore log collection")
-	} else {
-		err := findDatastore()
+	log.Info("vm-path not set, skipping datastore log collection")
+	// err := findDatastore()
```
Comment (sgairo) [inline cmd/vicadmin/logs.go]: Are we temporarily passing over the findDatastore() call or is it permanent?

File: cmd/vicadmin/vicadm.go [hunk 4]
```diff
@@ -288,6 +276,11 @@ func listVMPaths(ctx context.Context, s *session.Session) ([]logfile, error) {
 		return nil, err
 	}
 
+	self, err := guest.GetSelf(ctx, s)
+	if err != nil {
+		log.Errorf("Unable to get handle to self for log filetering")
```
Comment (sgairo) [inline cmd/vicadmin/vicadm.go]: "filtering"

File: cmd/vicadmin/logs.go (hunk 3 above)
Comment (sgairo) [inline cmd/vicadmin/logs.go]: Okay, I don't see a difference in the logs either I was just wondering what your thought process was in commenting versus removing it, thanks!

File: cmd/vicadmin/logs.go [hunk 5]
```diff
@@ -154,15 +154,12 @@ func configureReaders() map[string]entryReader {
...
 
-		if err != nil {
-			log.Warning(err)
-		}
-	}
+	// if err != nil {
```
Comment (hmahmood) [inline cmd/vicadmin/logs.go]: Left over?

**Comment by dougm** [comment]
lgtm

**Comment by hickeng** [comment]
poke

**Comment by hickeng** [comment]
LGTM (for hasan's LGTM via github that LGTM plugin doesn't recognize as an LGTM)

**Comment by hickeng** [comment]
poke

**Comment by hickeng** [comment]
poke

**Pull request merged**